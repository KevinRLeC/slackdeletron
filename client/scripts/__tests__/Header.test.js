import React from 'react';
import { shallow, mount, render } from 'enzyme';

import Header from '../Components/Header';

describe('<Header />', () => {
  it('renders a Header component in the loggedOut State', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('a').text()).toBe('Login With Slack');
  });

  it('renders a Header component in the loggedIn State', () => {
    const wrapper = shallow(<Header isLoggedIn />);
    expect(wrapper.find('a').text()).toBe('Logout');
  });
});
