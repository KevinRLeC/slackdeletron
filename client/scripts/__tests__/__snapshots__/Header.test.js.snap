// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Renders Correctly 1`] = `
<header
  className="Header"
>
  <h1
    className="Header__Logo"
  >
    Slack Deletron
  </h1>
  <div
    className="Header__Meta"
  >
     
    <a
      className="LoginButton"
      href="https://slack.com/oauth/authorize?client_id=123456789.987654321&scope=identity.basic"
    >
      <img
        alt="Sign in with Slack"
        height="40"
        src="https://platform.slack-edge.com/img/sign_in_with_slack.png"
        srcSet="https://platform.slack-edge.com/img/sign_in_with_slack.png 1x, https://platform.slack-edge.com/img/sign_in_with_slack@2x.png 2x"
        width="172"
      />
    </a>
  </div>
</header>
`;
